app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -286,7 +286,7 @@ def vms
 
     collector.vms.each do |s|
       if hosts && !s.os_ext_srv_attr_host.blank?
-        parent_host = hosts.find_by('lower(hypervisor_hostname) = ?', s.os_ext_srv_attr_host.downcase)
+        parent_host = hosts.find_by('lower(hypervisor_hostname) = ?', s.os_ext_srv_attr_host.split('.').first.downcase)
```

Comment:
Maybe ```split(/\.[^0-9]/)```? (and IPv6 doesn't have dots)..

Reply:
Can you use `os_ext_srv_attr_hypervisor_hostname`?  Is that field available on the OSP server model from the API?

https://ask.openstack.org/en/question/50587/os-ext-srv-attrhypervisor_hostname-and-os-ext-srv-attrhost-difference/

Reply:
`os_ext_srv_attr_hypervisor_hostname` should be available in OSP running on top of libvirt/kvm. I am not sure we should generally suppose that this field is available when running OS on other hypervisor (as answer you linked states).

Docs https://developer.openstack.org/api-ref/compute/#show-server-details says that `host` is `The name of the compute host on which this instance is running.` and `hypervisor_hostname` is `The hypervisor host name provided by the Nova virt driver. For the Ironic driver, it is the Ironic node uuid.` That seems like it should be populated by other hypervisors than libvirt/kvm. Anyway in TripleO installed OSP using libvirt/kvm both `os_ext_srv_attr_hypervisor_hostname` and `os_ext_srv_attr_host` have the same value.

Hosts in ManageIQ got their `hypervisor_hostname` attribute from `name` of server in Nova from Undercloud (https://github.com/ManageIQ/manageiq-providers-openstack/blob/fe9a981fffdd20fb6c0fd40dd319b624f782cf12/app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb#L331) and this `name` is defined by TripleO Heat Templates used to deploy OSP.

So in the end we trust that TripleO (THT) sets the hostnames for compute nodes to the same value (almost - hostname vs fqdn) as their names in Undercloud Nova.